### ISSUE TITLE
Newest versions of `numpy` cause strange errors to be printed to stdout

### ISSUE THREAD

**Issue by nmdickson** [nmdickson/GCfit#168]
In testing the latest version of this package, I've noticed that, when running the latest (1.24.x) version of `numpy`, the fitting functions begin to print a number of strange error messages to stdout:

```bash
>>> GCfitter NGC0104 nested
rv_cb_arr is NULL
Call-back cb_fcn_in___user__routines failed.
rv_cb_arr is NULL
Call-back cb_fcn_in___user__routines failed.
...
```

It seems like this is being caused by `f2py`, somewhere in the ODE module?
Either way, I don't think this is on us. If this truly is just an expected error being printed all of a sudden, we could in the meantime require a slightly older version of `numpy`, but I don't think this is worth worrying about.

**Comment by nmdickson** [comment]
Actually, this may in reality be an an error, which was being hidden by the sampler catch for failed-to-converge models:

```bash
>>> m = gcfit.Model(W0=4, M=5e5, rh=6, observations=gcfit.Observations('NGC362'))

rv_cb_arr is NULL
Call-back cb_fcn_in___user__routines failed.
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/nolan/Research/dev/GCfit/gcfit/core/data.py", line 1133, in __init__
    raise err
  File "/home/nolan/Research/dev/GCfit/gcfit/core/data.py", line 1121, in __init__
    super().__init__(**self._limepy_kwargs)
  File "/home/nolan/Research/dev/limepy_git/limepy/limepy.py", line 173, in __init__
    self._poisson(True)
  File "/home/nolan/Research/dev/limepy_git/limepy/limepy.py", line 446, in _poisson
    sol.integrate(self.maxr)
  File "/home/nolan/Research/dev/renv/lib/python3.10/site-packages/scipy/integrate/_ode.py", line 431, in integrate
    self._y, self.t = mth(self.f, self.jac or (lambda: None),
  File "/home/nolan/Research/dev/renv/lib/python3.10/site-packages/scipy/integrate/_ode.py", line 1174, in run
    x, y, iwork, istate = self.runner(*((f, t0, y0, t1) +
ValueError: setting an array element with a sequence. The requested array has an inhomogeneous shape after 1 dimensions. The detected shape was (22,) + inhomogeneous part.
```

Which seems like an error in `limepy` caused by the ODE integration in scipy.
This same model converges fine when using `numpy==1.23.x` (irrespective of the `scipy` version), so something must have changed in either the integration functions of the arrays that are passed into it.

**Comment by nmdickson** [comment]
I believe this has been fixed in the latest version of `limepy` ([1af3490](https://github.com/mgieles/limepy/commit/1af349010a3d404bc1ddb4dca7c0bbe5f40b2e1d)).
Fully updating `limepy`, `numpy` and `scipy` seems to be enough now to eliminate all warnings and errors.

Closing for now.
